lxqt-base/liblxqt/liblxqt-0.12.0.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit cmake-utils
+
+DESCRIPTION="Common base library for the LXQt desktop environment"
+HOMEPAGE="http://lxqt.org/"
+
+if [[ ${PV} = *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/lxde/${PN}.git"
+else
+	SRC_URI="https://github.com/lxde/${PN}/releases/download/${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+fi
+
+LICENSE="GPL-2 LGPL-2.1+"
+SLOT="0"
+
+RDEPEND=">=dev-libs/libqtxdg-3.1.0:=
+	dev-qt/qtcore:5=
+	dev-qt/qtdbus:5=
+	dev-qt/qtgui:5=
+	dev-qt/qtwidgets:5=
+	dev-qt/qtx11extras:5=
+	dev-qt/qtxml:5=
+	kde-frameworks/kwindowsystem:5=[X]
+	x11-libs/libXScrnSaver
+	!!lxqt-base/lxqt-common
```

Comment:
I tried `!`-mask, but it didn't work for me. Portage complained about file collisions and didn't merge new packages. So I made a hard mask.

Reply:
`!` should definitely prevent file collisions, did you test that with running `ebuild` or `emerge`?

Reply:
I've tested it with `emerge -avuDN system world`. I guess issue might be related to the fact that there are 3 packages which conflict with lxqt-common.

Reply:
then all three packages need the blocker against it and it will work.

Reply:
Of course I added the block in all packages with file collisions. Still didn't work for me last time I checked. And I don't want to downgrade it all and upgrade again just to ensure that it still doesn't work for me.

Reply:
The way it was done in my fork works. emerge was complaining about file colision, and I added the `!`-mask to fix them. It was installing without error afterward.

Reply:
Well, as I already told you, it didn't work same way for me for some reason.

Reply:
Did you check the file collision was not with a different package? Maybe you built with an outdated branch? `!!` is unacceptable for file collisions.

Reply:
Yes, I did check it.
If I currently try to install lxqt-common-0.11.2, I'm getting file collisions with 3 packages:
```
 * lxqt-base/lxqt-session-0.12.0:0::dt-overlay-patches
 *      /usr/bin/startlxqt
 *      /usr/share/kdm/sessions/lxqt.desktop
 *      /usr/share/man/man1/startlxqt.1.bz2
 *      /usr/share/xsessions/lxqt.desktop
 * 
 * x11-themes/lxqt-themes-0.12.0:0::dt-overlay-patches
 *      /usr/share/icons/hicolor/scalable/places/start-here-lxqt.svg
 *      /usr/share/lxqt/themes/ambiance/Butterfly-Kenneth-Wimer.jpg
 *      /usr/share/lxqt/themes/ambiance/arrow-left.svg
 *      /usr/share/lxqt/themes/ambiance/arrow-right.svg
 *      /usr/share/lxqt/themes/ambiance/lxqt-notificationd.qss
 *      /usr/share/lxqt/themes/ambiance/lxqt-notificationd/cancel.svg
 *      /usr/share/lxqt/themes/ambiance/lxqt-panel.qss
 *      /usr/share/lxqt/themes/ambiance/lxqt-runner.qss
 *      /usr/share/lxqt/themes/ambiance/lxqt-runner/clear.svg
 *      /usr/share/lxqt/themes/ambiance/lxqt-runner/navigation-menu.svg
 *      /usr/share/lxqt/themes/ambiance/mainmenu.svg
 *      /usr/share/lxqt/themes/ambiance/preview.png
 *      /usr/share/lxqt/themes/ambiance/wallpaper.cfg
 *      /usr/share/lxqt/themes/dark/arrow-right.svg
 * 
 * lxqt-base/lxqt-panel-0.12.0-r1:0::dt-overlay-patches
 *      /usr/share/desktop-directories/lxqt-leave.directory
 *      /usr/share/desktop-directories/lxqt-settings.directory
 * 
 * Package 'lxqt-base/lxqt-common-0.11.2' NOT merged due to file
 * collisions. If necessary, refer to your elog messages for the whole
 * content of the above message.
```

If I do downgrade and then try to upgrade, I'll hit these collisions, but this time those 3 packages would fail to merge due to collisions with lxqt-common. And no, while `!`-blocking should work, it didn't work for me.

Reply:
> EDIT: I suspect !-blocking may perfectly work for 1 package

Having dealt with dozens of file collisions in the past, I can assure you `!` works fine.

Reply:
I just merged your branch and modified to have `!`-blocks, and it worked! I was able to install lxqt-meta-0.11.1, and then lxqt-meta-0.12.0, and back to 0.11.1.

https://github.com/bassdr/gentoo/compare/iDarkTemplar/lxqt_updates...bassdr:lxqt_0.12
You might wanna simply merge [bassdr/lxqt_0.12](https://github.com/bassdr/gentoo/tree/lxqt_0.12) in your branch, but maybe you'll have to rebase on [gentoo/master](https://github.com/gentoo/gentoo) first.